monolith/views/users.py

```diff
@@ -30,27 +30,32 @@ def wall(author_id):
 @users.route('/wall/<author_id>/follow', methods=['GET'])
 @login_required
 def follow(author_id):
-    db.session.add(Follow(author_id, current_user.id))
     message = ''
-    try:
-        db.session.commit()
-        message = "Following!"
-    except IntegrityError:
-        message = "Already following!"
-    
+    if int(author_id)==current_user.id:
+        message= "Cannot follow yourself"
+    else:
+        db.session.add(Follow(author_id, current_user.id))
+        try:
+            db.session.commit()
+            message = "Following!"
+        except IntegrityError:
+            message = "Already following!"
     return render_template('follow.html', message = message)
 
 @users.route('/wall/<author_id>/unfollow', methods=['GET'])
 @login_required
 def unfollow(author_id):
-    msg = ""
-    if isFollowing(author_id, current_user.id) :
-        Follow.query.filter(Follow.user_id == author_id, Follow.followed_by_id == current_user.id).delete()
-        db.session.commit()
-        msg = "Unfollowed!"
+    message = ''
+    if int(author_id)==current_user.id:
```

Comment:
Now that I notice it, you can just tag author_id as int inside the route ('/wall/<int:author_id>/unfollow') instead of constructing an int here.

Reply:
Fixed